packages/mui-base/src/ListboxUnstyled/defaultListboxReducer.test.ts

```diff
@@ -0,0 +1,316 @@
+        selectedValue: null,
+      };
+
+      const action: ListboxAction<string> = {
+        type: ActionTypes.blur,
+        event: {} as any, // not relevant
+        props: {
+          options: [],
+          disableListWrap: false,
+          disabledItemsFocusable: false,
+          isOptionDisabled: () => false,
+          optionComparer: (o, v) => o === v,
+          multiple: false,
+        },
+      };
+
+      const result = defaultReducer(state, action);
+      expect(result.highlightedIndex).to.equal(-1);
+    });
+  });
+
+  describe('action: optionClick', () => {
+    it('sets the selectedValue to the clicked value', () => {
+      const state: ListboxState<string> = {
+        highlightedIndex: 42,
+        selectedValue: null,
+      };
+
+      const action: ListboxAction<string> = {
+        type: ActionTypes.optionClick,
+        event: {} as any, // not relevant
+        props: {
+          options: ['one', 'two', 'three'],
+          disableListWrap: false,
+          disabledItemsFocusable: false,
+          isOptionDisabled: () => false,
+          optionComparer: (o, v) => o === v,
+          multiple: false,
+        },
+        option: 'two',
+      };
+
+      const result = defaultReducer(state, action);
+      expect(result.selectedValue).to.equal('two');
+    });
+
+    it('add the clicked value to the selection if selectMultiple is set', () => {
+      const state: ListboxState<string> = {
+        highlightedIndex: 42,
+        selectedValue: ['one'],
+      };
+
+      const action: ListboxAction<string> = {
+        type: ActionTypes.optionClick,
+        event: {} as any, // not relevant
+        props: {
+          options: ['one', 'two', 'three'],
+          disableListWrap: false,
+          disabledItemsFocusable: false,
+          isOptionDisabled: () => false,
+          optionComparer: (o, v) => o === v,
+          multiple: true,
+        },
+        option: 'two',
+      };
+
+      const result = defaultReducer(state, action);
+      expect(result.selectedValue).to.deep.equal(['one', 'two']);
+    });
+
+    it('remove the clicked value from the selection if selectMultiple is set and it was selected already', () => {
+      const state: ListboxState<string> = {
+        highlightedIndex: 42,
+        selectedValue: ['one', 'two'],
+      };
+
+      const action: ListboxAction<string> = {
+        type: ActionTypes.optionClick,
+        event: {} as any, // not relevant
+        props: {
+          options: ['one', 'two', 'three'],
+          disableListWrap: false,
+          disabledItemsFocusable: false,
+          isOptionDisabled: () => false,
+          optionComparer: (o, v) => o === v,
+          multiple: true,
+        },
+        option: 'two',
+      };
+
+      const result = defaultReducer(state, action);
+      expect(result.selectedValue).to.deep.equal(['one']);
+    });
+  });
+
+  describe('action: keyDown', () => {
+    describe('Home key is pressed', () => {
```

Comment:
Can we also add a test to verify that it works if the first item is a group title?